apps/app/src/electron/HTTPAPI.ts

```diff
@@ -0,0 +1,134 @@
+import Koa from 'koa'
+import Router from '@koa/router'
+import { IPCServer, isUndoable } from './IPCServer'
+import { stringifyError } from '@shared/lib'
+
+export const HTTP_API_PORT = 5500
+
+export class HTTPAPI {
+	private app = new Koa()
+	private router = new Router()
+
+	private methodSignatures: {
+		[fullEndpoint: string]: {
+			endpoint: string
+			type: string
+		}
+	} = {}
+
+	constructor(port: number, ipcServer: IPCServer) {
+		this.router.get(`/`, async (ctx) => {
+			ctx.response.body = `<html><body>
+			<a href="/api">API</a>
+			</body></html>`
+			ctx.response.status = 200
+		})
+		this.router.get(`/api`, async (ctx) => {
+			const methods = Object.entries(this.methodSignatures)
+				.map(([_fullEndpoint, e]) => {
+					const url = `/api/${e.endpoint}/?`
+
+					return `<a href="${url}">${e.type} ${url}</a>`
+				})
+				.join('<br>\n')
+
+			ctx.response.body = `<html><body>${methods}</body></html>`
+			ctx.response.status = 200
+		})
+
+		for (const methodName of Object.getOwnPropertyNames(IPCServer.prototype)) {
+			// Ignore "private" methods.
+			if (methodName[0] === '_') continue
+			if (methodName === 'constructor') continue
+
+			const originalMethod = (ipcServer as any)[methodName]
+			// Ignore methods that don't exist.
+			if (!originalMethod) continue
+
+			const fcn = originalMethod.bind(ipcServer)
+
+			let endpoint: string
+			let endpointType: 'GET' | 'POST' | 'DELETE'
+
+			if (methodName.startsWith('get')) {
+				// Handle GET requests
+
+				endpoint = methodName.charAt(3).toLocaleLowerCase() + methodName.substring(4)
+				endpointType = 'GET'
+				this.router.get(`/api/${endpoint}`, async (ctx) => {
+					try {
+						const result = await fcn(ctx.request.query)
+						ctx.response.body = result
+						ctx.response.status = 200
+					} catch (error) {
+						const stringifiedError = stringifyError(error)
+						if (stringifiedError.includes('not found')) {
```

Comment:
It is slightly better to to `.match(/not found/i)` to be case insensitive

Reply:
Resolved in https://github.com/SuperFlyTV/SuperConductor/pull/83/commits/971741312f0e4235319e0ebf471d6211cb839d54